pubtools/_quay/push_docker.py

```diff
@@ -422,6 +417,16 @@ def run(self):
             LOG.error("An exception has occurred during the push, starting rollback")
             self.rollback(backup_tags, rollback_tags)
             raise e
+        else:
+            # Remove old signatures
+            container_signature_handler.remove_outdated_signatures(
+                [
+                    (image_data.repo, manifest["digest"], image_data.tag)
```

Comment:
Are you getting these digests correctly? The manifest will be either manifest list in which case you need to iterate over "manifests" to get all digests. Or (in case of source image) it will be a V2S2 manifest which doesn't contain digest data. You may use the same approach as "generate_backup_mapping" to gather this digest.